### ISSUE TITLE
https support?

### ISSUE THREAD

**Issue by rieje** [akrennmair/newsbeuter#552]
Newsbeuter can't use https instead of http to gather necessary data? Isn't this a security/privacy risk?

Or maybe Tiny Tiny RSS does and I can use that as the backend for newsbeuter.

**Comment by lutzhorn** [comment]
Which HTTPS URL does Newsbeuter not fetch? There are many feed URLs in my list Newsbeuter handles without any problem.

**Comment by rieje** [comment]
Not the feeds itself, but the checking if there is up-to-date stuff is done via HTTP because it's quicker, according to the man page.

**Comment by lutzhorn** [comment]
> checking if there is up-to-date stuff is done

What do you mean by that? What exact operation do you refer to? Can you please cite the relevant section of the man page?

**Comment by rieje** [comment]
>  Newsbeuter also uses a number of measures to preserve the users' and feed providers' bandwidth, by trying to avoid
       unnecessary feed downloads through the use of conditional HTTP downloading. It saves every feed’s "Last-Modified" and
       "ETag" response header values (if present) and advises the feed’s HTTP server to only send data if the feed has been
       updated by modification date/time or "ETag" header. This doesn’t only make feed downloads for RSS feeds with no new
       updates faster, it also reduces the amount of transferred data per request. Conditional HTTP downloading can be
       optionally disabled per feed by using the "always-download" configuration command.

**Comment by Minoru** [comment]
HTTPS isn't a whole new protocol; it's just good old HTTP wrapped into a TLS tunnel. `ETag` and `Last-Modified` headers are part of HTTP, so they work with HTTPS too.

Newsbeuter decides what protocol to use based on the feed's URL alone. If the URL starts with "http://", HTTP will be used; if it starts with "https://", HTTPS will be used. `always-download` doesn't affect this decision; the setting only disables the aforementioned headers.

In short, Newsbeuter is going to use HTTPS exclusively if all your feed URLs start with "https://".

If that answers your question, please close the issue.

**Comment by Minoru** [comment]
And @lutzhorn—thanks a lot for your comments! It's nice to see the problem solved before I even had the chance to read the first comment.

**Comment by lutzhorn** [comment]
@rieje For HTTP Conditional GET you can also take a look at https://developer.mozilla.org/en-US/docs/Web/HTTP/Conditional_requests.

It is not the use of TLS (the "S" in HTTPS) that is conditional, but the decision of the server to actually return a response body is based on conditions given by the client in the request.
